src/media-segment-request.js

```diff
@@ -0,0 +1,281 @@
+import videojs from 'video.js';
+import { createTransferableMessage } from './bin-utils';
+
+export const REQUEST_ERRORS = {
+  ABORTED: -101,
+  TIMEOUT: -102,
+  FAILURE: 2
+};
+
+/**
+ * Turns segment byterange into a string suitable for use in
+ * HTTP Range requests
+ */
+const byterangeStr = function(byterange) {
+  let byterangeStart;
+  let byterangeEnd;
+
+  // `byterangeEnd` is one less than `offset + length` because the HTTP range
+  // header uses inclusive ranges
+  byterangeEnd = byterange.offset + byterange.length - 1;
+  byterangeStart = byterange.offset;
+  return 'bytes=' + byterangeStart + '-' + byterangeEnd;
+};
+
+/**
+ * Defines headers for use in the xhr request for a particular segment.
+ */
+const segmentXhrHeaders = function(segment) {
+  let headers = {};
+
+  if (segment.byterange) {
+    headers.Range = byterangeStr(segment.byterange);
+  }
+  return headers;
+};
+
+const abortAll = (activeXhrs) => {
+  Object.keys(activeXhrs).forEach((xhrKey) => {
+    const xhr = activeXhrs[xhrKey];
+
+    // only abort xhrs that haven't had a response
+    if (!xhr.responseTime) {
```

Comment:
I haven't seen any issues. Aborting an already aborted xhr seems to be a no-op and doesn't trigger any sort of readystatechange.